src/frame/headers.rs

```diff
@@ -892,6 +901,8 @@ impl HeaderBlock {
 
                         headers_size += decoded_header_size(name.as_str().len(), value.len());
                         if headers_size < max_header_list_size {
+                            self.field_size +=
+                                decoded_header_size(name.as_str().len(), value.len());
```

Comment:
Should we check that `decoded_header_size` won't exceed `max_header_list_sized`?

Reply:
that is covered under other checks elsewhere

Reply:
Nvm, that's exactly what this does

Reply:
Yeah I don't know why I said that lol, it's literally checked right here.